### PR TITLE
To support multiple architectures as in MinIO Repo

### PR DESCRIPTION
**Pull request by cniackz**
### Objective:

To push Helm Operator Image to Quay Repo. This time using multiple Architectures as in MinIO Binary: https://github.com/minio/minio/blob/master/docker-buildx.sh#L7 where multiple architectures are observed and same would be expected for this image:

* Link: https://quay.io/repository/minio/minio/manifest/sha256:6fce70c2fd719603cb538e25ec3f58d77c12c1f5141e46240a850065c7ce4470

Screenshot 2023-09-06 at 7 06 17 AM

* Link: https://quay.io/repository/operator-framework/helm-operator/manifest/sha256:4c1f130f386aa084efa00755bff07a3d41b76885efdb5f2e2bd33819d2083e8b

Screenshot 2023-09-06 at 7 13 58 AM

### For Admin:

* This is how you can push the image:

```sh
cd ~/directpv/helm

docker buildx prune -f

docker buildx build --push --no-cache \
    -t quay.io/minio/directpv-operator:4.0.7 \
    --platform=linux/amd64,linux/arm64,linux/ppc64le,linux/s390x \
    -f Dockerfile .
```

### CODE REVIEW

File: helm/generate.sh [hunk 1]
```diff
@@ -46,12 +46,12 @@ EOF
...
-    # docker-build: Build docker image with the manager.
-    docker buildx build --platform linux/amd64 -t "${IMG}" .
-
     # Ask admin to run this command for you as it requires privileges
-    # docker-push: Push docker image with the manager.
-    # docker push "${IMG}"
+    # docker-build: Build and push docker image with the manager.
+    docker buildx build --push --no-cache \
+    -t "${IMG}" \
+    --platform=linux/amd64,linux/arm64,linux/ppc64le,linux/s390x \
```
Comment (balamurugana) [inline helm/generate.sh:53]: This won't work. We should check whether sidecars are available for these architectures or not.

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: helm/generate.sh (hunk 1 above)
Comment (cniackz) [reply helm/generate.sh:53]: Bala the manager binary from operator-framework already supports it:

https://quay.io/repository/operator-framework/helm-operator/manifest/sha256:4c1f130f386aa084efa00755bff07a3d41b76885efdb5f2e2bd33819d2083e8b

* linux | amd64
* linux | arm64
* linux | ppc64le
* linux | s390x

What do you mean by sidecars?

Comment (balamurugana) [reply helm/generate.sh:53]: Please refer https://github.com/minio/directpv/blob/master/docs/installation.md#prerequisites-1. Without these k8s sidecars, DirectPV doesn't work.

Comment (cniackz) [reply helm/generate.sh:53]: You are right @balamurugana closing this PR then 👍

**Pull request closed without merging**